lib/BGA.tsx

```diff
@@ -0,0 +1,89 @@
+import { Cuboid, Sphere, Translate, Colorize, Rotate } from "jscad-fiber";
+import { fp } from "@tscircuit/footprinter";
+import { useMemo } from "react";
+
+interface BGAProps {
+  packageWidth?: number;
+  packageLength?: number;
+  packageHeight?: number;
+  standoffHeight?: number;
+  ballPitch?: number;
+  ballDiameter?: number;
+  ballRows?: number;
+  ballColumns?: number;
+  missingBalls?: number[];
+  footprintString?: string;
+}
+
+export const BGA = ({
+  packageWidth = 10,
+  packageLength = 10,
+  packageHeight = 1.2,
+  standoffHeight = 0.2,
+  ballPitch = 0.8,
+  ballDiameter = 0.5,
+  ballRows = 8,
+  ballColumns = 8,
+  missingBalls = [],
+  footprintString,
+}: BGAProps) => {
+  const bodyHeight = packageHeight - standoffHeight;
+  const bodyOffset = 1.3; 
+  const ballOffset = 0;
+
+  const ballsSoup = useMemo(() => {
+    if (!footprintString) return null;
+    const result = fp.string(footprintString);
+    return result.soup();
+  }, [footprintString]);
+
+  return (
+    <>
+      {/* Package body */}
+      <Colorize color="#555">
+        <Rotate rotation={[190, 0, 0]}>
```

Comment:
```suggestion
        <Rotate rotation={[180, 0, 0]}>
```

Reply:
We discussed on discord that this is maybe an issue with `jscad-fiber` where we need to make things multiples of `Math.PI`/`Math.PI/2`